### PR TITLE
Feature/create react components

### PR DESCRIPTION
**Pull request by mayurkamaliya**
Created components,
1. Counter
A component that renders a numeric value and two buttons: one to decrement the value by 1, another to increment the value by 1.
The component should take an initial value in a property.
The component should be written using EcmaScript classes syntax, the `render` method should use `React.createElement` API (without JSX).

2. SearchForm
The component should accept two properties:
Initial search query. Use the value to set the initial value for the input
A "onSearch" callback property. Call the callback property every time the user presses Enter when the input has focus or when the user clicks the Search button. Pass current input value in callback arguments.

3. GenreSelect
A list of genre names to display. Use the incoming list to render genre buttons.
A name of currently selected genre. Use the name to identify which button to highlight.
A "onSelect" callback property. Call the callback function when the user clicks on any genre button. Pass respective genre name to the callback arguments.

## Visuals:
2023-10-09 10_55_32-Settings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-    "react-scripts": "5.0.1",
+    "react-scripts": "^3.0.1",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "start": "react-scripts --openssl-legacy-provider start",
```
Comment (andreyBalabukha) [inline package.json:15]: Why did you add `--openssl-legacy-provider`?

File: src/components/searchform.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  handleKeyPress = (e) => {
+    if (e.key === 'Enter') {
+      this.handleSearch();
+    }
+  };
+
+  render() {
+    return (
+      <div className="form-container">
+        <p className="header-element">FIND YOUR MOVIE</p>
```
Comment (andreyBalabukha) [inline src/components/searchform.js:29]: Find Your Movie - please move all the literals to a separate file, like literals.js
for example:
`export const FIND_YOUR_MOVIE = 'Find Your Movie'`

File: src/css/genreselect.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  .red {
+    margin: 5px;
+    padding: 10px;
+    font-size: 23px;
+    color: rgb(201, 47, 47);
+    cursor: pointer;
+    border: none;
+    background-color: transparent;
+    font-variant-caps: all-petite-caps;
+  }
```
Comment (andreyBalabukha) [inline src/css/genreselect.css:27]: Please add a blank line

File: src/css/searchform.css [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  
+  .form-button {
+    margin: 5px;
+    padding: 10px;
+    font-size: 16px;
+    border-radius: 50px;
+    background-color:  rgb(202, 146, 140);
+    text-emphasis-color: inherit;
+    cursor: pointer;
+  }
```
Comment (andreyBalabukha) [inline src/css/searchform.css:31]: Please add a blank line

**Review by andreyBalabukha — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mayurkamaliya) [reply package.json:15]: Without this option, it gives me the below error
image

File: src/components/searchform.js (hunk 2 above)
Comment (mayurkamaliya) [reply src/components/searchform.js:29]: Created literlas.js and exported the constant.

File: src/css/searchform.css (hunk 4 above)
Comment (mayurkamaliya) [reply src/css/searchform.css]: Added blank line.

File: src/css/genreselect.css (hunk 3 above)
Comment (mayurkamaliya) [reply src/css/genreselect.css]: Added blank line.

**Review by andreyBalabukha — APPROVED** [review]

**Review by andreyBalabukha — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (andreyBalabukha) [reply package.json:15]: ok, let's proceed with your solution.
